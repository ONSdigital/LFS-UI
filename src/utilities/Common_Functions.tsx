function weeks() {
    let i;
    let weekList = []
    for (i=1; i<=52; i++) {
        weekList.push({"label":"Week "+String(i), "value":String(i)})
    }
    return weekList
}

function months() {
    const monthNames = ["January", "February", "March", "April", "May", "June", "July", 
                        "August", "September", "October", "November", "December"];
    let i;
    let monthlist = [];
    var date = new Date();
    const month = date.getMonth();
    for (i=0; i<12; i++){
        if (month-i>=0)
            monthlist.push({"label":monthNames[month-i], "value":monthNames[month-i]})
        else{
            monthlist.push({"label":monthNames[month+12-i], "value":monthNames[month+12-i]})

        }
    }
    return monthlist
}

const quarters = [{"label":"Q1", "value":"Q2"}, 
                  {"label":"Q2", "value":"Q2"}, 
                  {"label":"Q3", "value":"Q3"}, 
                  {"label":"Q4", "value":"Q4"}]

function years() {
    let i;
    let yearlist = [];
    var date = new Date(); 
    const year = date.getFullYear();
    for (i=0; i<10; i++){
        yearlist.push({"label":String(year-i), "value":String(year-i)})
    }
    return yearlist
}

function getMonth (Batch_ID: String) {
    var Month = Batch_ID.substring(0,3)
    switch (Month){
      case 'Jan': Month = 'January'; break;
      case 'Feb': Month = 'February'; break;
      case 'Mar': Month = 'March'; break;
      case 'Apr': Month = 'April'; break;
      case 'May': break;
      case 'Jun': Month = 'June'; break;
      case 'Jul': Month = 'July'; break;
      case 'Aug': Month = 'August'; break;
      case 'Sep': Month = 'September'; break;
      case 'Oct': Month = 'October'; break;
      case 'Nov': Month = 'November'; break;
      case 'Dec': Month = 'December'; break;
    }
    return Month
  }

  function getYear (Batch_ID: String) {
    return Batch_ID.substring(3,7)
  }

const batches = [{"label":"Monthly", "value":"monthly"}, 
        {"label":"Quarterly", "value":"quarterly"}, 
        {"label":"Yearly", "value":"yearly"}]

const qList = ["January", "February", "April", "May", "July", "August", "October", "November"]

export{weeks, months, quarters, years, batches, getMonth, getYear, qList}