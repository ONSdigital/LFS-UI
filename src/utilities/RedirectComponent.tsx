import {useHistory} from "react-router-dom";
import {useEffect} from "react";

interface Props {
    url: string
}

export function RedirectComponent(props : Props) {
    let history = useHistory();

    useEffect(() => {
        history.push(props.url);
    });
}