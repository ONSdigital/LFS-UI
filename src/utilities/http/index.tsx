export * from "./survey";
export * from "./http";
export * from './batch';