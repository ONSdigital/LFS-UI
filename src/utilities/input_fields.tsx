function weeks() {
    let i;
    let weekList = []
    for (i=1; i<=52; i++) {
        weekList.push({"label":"Week "+String(i), "value":String(i)})
    }
    return weekList
}

function months() {
    const monthNames = ["January", "February", "March", "April", "May", "June", "July", 
                        "August", "September", "October", "November", "December"];
    let i;
    let monthlist = [];
    var date = new Date();
    const month = date.getMonth();
    for (i=0; i<12; i++){
        if (month-i>=0)
            monthlist.push({"label":monthNames[month-i], "value":monthNames[month-i]})
        else{
            monthlist.push({"label":monthNames[month+12-i], "value":monthNames[month+12-i]})

        }
    }
    return monthlist
}

const quarters = [{"label":"Q1", "value":"Q2"}, 
                  {"label":"Q2", "value":"Q2"}, 
                  {"label":"Q3", "value":"Q3"}, 
                  {"label":"Q4", "value":"Q4"}]

function years() {
    let i;
    let yearlist = [];
    var date = new Date(); 
    const year = date.getFullYear();
    for (i=0; i<10; i++){
        yearlist.push({"label":String(year-i), "value":String(year-i)})
    }
    return yearlist
}

const runs = [{"label":"Monthly", "value":"monthly"}, 
        {"label":"Quarterly", "value":"quarterly"}, 
        {"label":"Yearly", "value":"yearly"}]

export{weeks, months, quarters, years, runs}