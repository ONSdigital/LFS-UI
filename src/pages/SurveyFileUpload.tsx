import React, {Component} from "react";
import {ONSUpload} from "../components/ONS_DesignSystem/ONSUpload";
import {ONSButton} from "../components/ONS_DesignSystem/ONSButton";
import {getSurveyAudit, postSurveyFile} from "../utilities/http";
import {ONSPanel} from "../components/ONS_DesignSystem/ONSPanel";
import {ONSMetadata} from "../components/ONS_DesignSystem/ONSMetadata";
import {getStatusStyle, monthNumberToString} from "../utilities/Common_Functions";
import {ONSAccordionTable} from "../components/ONS_DesignSystem/ONSAccordionTable";
import {ONSStatus} from "../components/ONS_DesignSystem/ONSStatus";
import {SURVEY_UPLOAD_HISTORY} from "../utilities/Headers";
import DocumentTitle from "react-document-title";
import {FileUploadProgress} from "./FileUploadProgress";
import {Link} from "react-router-dom";
import dateFormatter from "dayjs";
import {ONSBreadcrumbs} from "../components/ONS_DesignSystem/ONSBreadcrumbs";

interface Props {
    period: string,
    year: string,
    surveyType: string
    match: any
}

interface Panel {
    label: string,
    visible: boolean
    status: string
}

interface State {
    fileOne: any,
    panel: Panel
    uploading: boolean,
    importStarted: boolean
    surveyType: string,
    year: string
    week: string
    month: string
    period: string
    uploadHistory: []
    importHidden: boolean
    importFileName: string
    metaData: any[]
}

export class SurveyFileUpload extends Component <Props, State> {
    displayName = SurveyFileUpload.name;

    constructor(props: Props) {
        super(props);
        let period = (props.match.params.survey === "gb" ? "Week " + props.match.params.week.toString() : monthNumberToString(Number(props.match.params.month)).toString());
        this.state = {
            fileOne: "",
            uploading: false,
            importStarted: false,
            surveyType: (props.match.params.survey).toUpperCase(),
            year: props.match.params.year,
            week: props.match.params.week,
            month: props.match.params.month,
            period: period,
            panel: {
                label: '',
                visible: false,
                status: ''
            },
            uploadHistory: [],
            importHidden: false,
            importFileName: props.match.params.survey + (props.match.params.survey === 'gb' ? props.match.params.week : props.match.params.month) + props.match.params.year,
            metaData:
                [
                    {
                        L: "Survey",
                        R: props.match.params.survey.toUpperCase(),
                    },
                    {
                        L: "Year",
                        R: props.match.params.year.toString(),
                    },
                    {
                        L: "Period",
                        R: period,
                    }
                ]
        };

        this.handleFileChange = this.handleFileChange.bind(this);
    }

    componentDidMount(): void {
        this.getUploadHistory()
    }

    getUploadHistory = () => {
        getSurveyAudit(this.state.surveyType, this.state.year, (this.state.surveyType === 'GB' ? this.state.week : this.state.month))
            .then(r => {
                if (r !== undefined) {
                    // Batch does not exist, load not found page
                    this.setState({uploadHistory: r});
                }
            })
            .catch(error => {
                console.log(error);
            });
    };

    uploadFile = () => {
        console.log('Uploading File');
        this.setState({
            uploading: true,
            importStarted: true,
            panel: {
                label: '',
                visible: false,
                status: ''
            }
        });

        if (this.state.fileOne.length === 0) {
            this.setState({
                uploading: false,
                panel: {
                    label: 'No File Selected',
                    visible: true,
                    status: 'info'
                }
            });
            return
        }

        postSurveyFile(this.state.fileOne, this.state.importFileName, 'survey', this.state.surveyType.toLowerCase(), (this.state.surveyType === "GB" ? this.state.week : this.state.month), this.state.year)
            .then(response => {
                console.log(response);
                if (response.status === 'ERROR') {
                    this.setPanel("Error Occurred while Uploading File: " + response.errorMessage.toString(), 'error', true);
                } else {
                    this.setPanel(this.state.surveyType.toUpperCase() + " Survey Uploaded, Starting Import", 'success', true);
                }
                this.setState({
                    uploading: false,
                });
            })
            .catch(err => {
                console.log(err);
                if (err.toString() === "SyntaxError: Unexpected token P in JSON at position 0") {
                    this.setPanel("Error Occurred while Uploading File: Unable to Connect to Server", 'error', true);
                } else {
                    this.setPanel("Error Occurred while Uploading File: " + err.toString(), 'error', true);
                }
                this.setState({
                    uploading: false,
                });
            });
    };

    returnToManageBatch = (notImported: boolean) => {
        let redirectURL = '';
        if (!notImported) {
            redirectURL = "/" + this.state.surveyType + '-' + this.state.week + '-' + this.state.month + '-' + this.state.year;
            if (!this.state.importStarted) {
                return
            }
        }
        window.location.href = "/manage-batch/monthly/" + this.state.year + "/" + this.state.month + redirectURL
    };

    setPanel = (message: string, status: string, visible: boolean) => {
        this.setState({
            panel: {
                label: message,
                visible: visible,
                status: status
            }
        });
        // if (message.indexOf("File Imported Successfully")) {
        //     this.returnToManageBatch(false)
        // }
    };

    setFileUploading = (bool: boolean) => {
        this.setState({importHidden: bool})
    };

    handleFileChange = (selectorFiles: FileList | null) => {
        console.log(selectorFiles);
        this.setState({fileOne: selectorFiles})
    };

    UploadHistoryRow = (rowData: any) => {
        let row = rowData.row;
        return (
            <>
                <td className="table__cell ">
                    {dateFormatter(row.uploadDate).format("DD/MM/YYYY HH:mm")}
                </td>
                <td className="table__cell ">
                    <ONSStatus label={row.message} small={false}
                               status={getStatusStyle(+row.status).colour}/>
                </td>
            </>
        )
    };

    render() {
        let batchLink = "manage-batch/monthly/" + this.state.year + "/" + this.state.month

        return (
            <DocumentTitle
                title={'LFS Survey Import ' + this.state.period + ' ' + this.state.year + ' ' + this.state.surveyType.toUpperCase()}>
                <div className="container">
                    <ONSBreadcrumbs List={[{name: "Home", link: ""}, {name: "Manage Batch " + monthNumberToString(Number(this.state.month)) + " " + this.state.year, link: batchLink}]}
                                    Current={"Import Survey - " + this.state.period} />
                    <h2>Import Survey</h2>
                    <ONSMetadata List={this.state.metaData}/>
                    <div style={{width: "55%"}}>
                        <h4>Previous imports </h4>
                        <ONSAccordionTable Headers={SURVEY_UPLOAD_HISTORY} data={this.state.uploadHistory}
                                           Row={this.UploadHistoryRow}
                                           expandedRowEnabled={false}
                                           noDataMessage={"Survey has not been previously imported"}/>
                    </div>
                    <ONSPanel status={this.state.panel.status} label={this.state.panel.label}
                              hidden={!this.state.panel.visible}>
                        <p>{this.state.panel.label}</p>
                    </ONSPanel>
                    <br/>
                    <form hidden={this.state.importHidden}>
                        <ONSUpload label={"Import " + this.state.surveyType.toUpperCase() + " File"}
                                   description={"Only .sav accepted"} fileName={"Upload 1"}
                                   fileID={"U1"}
                                   accept=".sav" onChange={(e) => this.handleFileChange(e.target.files)}/>
                        <br/>
                        <ONSButton id={'import-btn'} label={"Import"} field={true} onClick={this.uploadFile} primary={true} small={false}
                                   loading={this.state.uploading}/>
                        <Link className={'field'} style={{marginLeft: "15px"}}
                              to={"/manage-batch/monthly/" + this.state.year + "/" + this.state.month}>
                            <ONSButton label={"Return to Manage Batch"} field={true} primary={false}
                                       small={false}/>
                        </Link>
                    </form>
                    <br/>
                    <FileUploadProgress importName={this.state.surveyType.toUpperCase() + " Survey File"}
                                        fileName={this.state.importFileName} hidden={false}
                                        fileUploading={this.setFileUploading} setPanel={this.setPanel}
                                        redirectOnComplete={this.returnToManageBatch}/>
                    <br/>
                </div>
            </DocumentTitle>
        )
    }
}