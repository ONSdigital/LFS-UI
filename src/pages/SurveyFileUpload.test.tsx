import React from "react";

import {SurveyFileUpload} from "./SurveyFileUpload";

import {default as flushPromises} from "../tests/util/flushPromises";
import {cleanup, fireEvent, render, screen} from "@testing-library/react";

import {BrowserRouter} from "react-router-dom";

import Enzyme, {mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import fetch from "../tests/setup/__mocks__/fetch.js";

import {monthNumberToString} from "../utilities/Common_Functions";
import {act} from "react-dom/test-utils";
import WS from "jest-websocket-mock";
import MockDate from "mockdate";

// @ts-ignore
global.fetch = fetch;
MockDate.set('2015-01-21');


describe("GB and NI survey file uploads", () => {
    Enzyme.configure({adapter: new Adapter()});

    let server: WS;

    beforeEach(() => {
        server = new WS("ws://127.0.0.1:8000/ws", {jsonProtocol: true});
    });

    afterEach(() => {
        cleanup();
        server.close();
    });

    const match = {
        path: "/survey-import/:survey/:week/:month/:year",
        url: "/survey-import/gb/1/1/2019",
        isExact: true,
        params: {survey: "gb", week: "1", month: "1", year: "2019"}
    };

    const matchNI = {
        path: "/survey-import/:survey/:week/:month/:year",
        url: "/survey-import/ni/0/5/2019",
        isExact: true,
        params: {survey: "ni", week: "0", month: "5", year: "2019"}
    };

    const Props = {
        period: "Week 1",
        year: "2019",
        surveyType: "gb",
        match: match
    };

    const PropsNI = {
        period: 5,
        year: "2019",
        surveyType: "ni",
        match: matchNI
    };

    function wrapper(render: any, props: any) {
        return render(
            <BrowserRouter>
                <SurveyFileUpload {...props}/>
            </BrowserRouter>
        );
    }

    /*
    Function to Mock a file in the File Selection, then press the Submit button
     */
    async function selectAndImportFile(getByTestId: any, getByLabelText: any, fileName: string) {
        const inputEl = getByLabelText(/import/i);

        const file = new File(["(⌐□_□)"], fileName, {
            type: "sav"
        });

        Object.defineProperty(inputEl, "files", {
            value: [file]
        });

        fireEvent.change(inputEl);

        fireEvent.click(getByTestId("import-button"));

        await act(async () => {
            await flushPromises();
        });
    }

    it("renders the page", async () => {
        expect(wrapper(render, Props)).toMatchSnapshot();
    });

    it("displays the correct GB metadata", async () => {
        wrapper(render, Props);

        expect(screen.getByTestId(/metadata-test-0/i).textContent).toEqual(Props.surveyType.toUpperCase());
        expect(screen.getByTestId(/metadata-test-1/i).textContent).toEqual(Props.year.toUpperCase());
        expect(screen.getByTestId(/metadata-test-2/i).textContent).toEqual(Props.period);
    });

    it("displays the correct NI metadata", async () => {
        wrapper(render, PropsNI);

        expect(screen.getByTestId(/metadata-test-0/i).textContent).toEqual(PropsNI.surveyType.toUpperCase());
        expect(screen.getByTestId(/metadata-test-1/i).textContent).toEqual(PropsNI.year.toUpperCase());
        expect(screen.getByTestId(/metadata-test-2/i).textContent).toEqual(monthNumberToString(PropsNI.period));
    });

    test("breadcrumb points to the correct url", async () => {
        let breadWrapper = wrapper(mount, Props);

        //Home breadcrumb links to Home
        expect(breadWrapper.find("ONSBreadcrumbs").getElement().props.List[0].link).toEqual("");

        //manage batch breadcrumb links to the correct manage batch page
        expect(breadWrapper.find("ONSBreadcrumbs").getElement().props.List[1].link).toEqual("manage-batch/monthly/" + Props.year + "/" + Props.period.slice(-1));
    });

    it("displays file upload history in the table", async () => {
        const {getByTestId, getByText} = wrapper(render, Props);

        await act(async () => {
            await flushPromises();
        });

        expect(getByText(/Insert survey row failed/i)).toBeTruthy();
    });

    it("displays message in table when survey has not been imported before", async () => {
        const {getByTestId, getByText} = wrapper(render, PropsNI);

        await act(async () => {
            await flushPromises();
        });

        expect(getByText(/Survey has not been previously imported/i)).toBeTruthy();
    });

    it("displays \"no file selected\" panel when the upload doesnt contain a file", async () => {
        const {getByTestId, getByText} = wrapper(render, Props);

        fireEvent.click(getByTestId("import-button"));

        let panel = await getByTestId(/import=panel/i);
        expect(panel.textContent).toEqual("No File Selected");
    });

    it("selects a file and uploads, returning an OK response and displaying a success panel", async () => {
        const {getByTestId, getByLabelText, getByText} = wrapper(render, Props);

        await selectAndImportFile(getByTestId, getByLabelText, "GB_File.sav");

        let panel = await getByTestId(/import=panel/i);
        expect(panel.textContent).toContain("Survey Uploaded, Starting Import");
    });

    it("selects a file and fails to upload, returning an ERROR response and displaying a error panel", async () => {
        const {getByTestId, getByLabelText, getByText} = wrapper(render, Props);

        await selectAndImportFile(getByTestId, getByLabelText, "GB_File_invalid_file.sav");

        let panel = await getByTestId(/import=panel/i);
        expect(panel.textContent).toEqual("Error Occurred while Uploading File: Error Occurred");
    });

    it("selects a file and it can't connect to the server, it should display the error panel", async () => {
        const {getByTestId, getByLabelText, getByText} = wrapper(render, Props);

        await selectAndImportFile(getByTestId, getByLabelText, "GB_File_server_not_online.sav");

        let panel = await getByTestId(/import=panel/i);
        expect(panel.textContent).toEqual("Error Occurred while Uploading File: Unable to Connect to Server");
    });

    test("when an import is successful the page should redirect back to the Manage Batch page with the summary", async () => {
        delete window.location;
        // @ts-ignore - added so it doesnt have all other unused parameters
        window.location = {href: "/survey-import/gb/1/1/2019"};

        server.on("connection", socket => {
            console.log("Server Connection");

            socket.on("message", function incoming(message) {
                console.log("Received from client:  %s", message);
                server.send({fileName: "file12", percent: 100, status: 2, errorMessage: ""});
            });
        });

        const {getByTestId, getByLabelText, getByText} = wrapper(render, Props);

        await server.connected;

        await selectAndImportFile(getByTestId, getByLabelText, "GB_File.sav");

        // Send a Import Complete status
        await server.send({fileName: "file12", percent: 100, status: 2, errorMessage: ""});

        await flushPromises();

        expect(window.location.href).toEqual("/manage-batch/monthly/2019/1/GB-1-1-2019");
        server.close();
    });

    test("when you load the page and a import is successful but an new Import is not been started, it should NOT redirect back to the Manage Batch page", async () => {
        delete window.location;
        // @ts-ignore - added so it doesnt have all other unused parameters
        window.location = {href: "/survey-import/gb/1/1/2019"};

        /*:
        On connection to the server it will return a mock response with the status of a file.
        This would usually be returned after the message is sent from the client (UI)
        but here it is returned immediately on connection for the test
        */
        server.on("connection", _ => {
            server.send({fileName: "file12", percent: 100, status: 2, errorMessage: ""});
            // socket.close({ wasClean: false, code: 1003, reason: "NOPE" });
        });

        const {getByTestId, getByLabelText, getByText} = wrapper(render, Props);

        await server.connected;

        await flushPromises();

        // Make sure the page has not changed
        expect(window.location.href).toEqual("/survey-import/gb/1/1/2019");
    });

    it("selects a file and an unknown error is returned, display error", async () => {
        const {getByTestId, getByLabelText, getByText} = wrapper(render, PropsNI);

        await selectAndImportFile(getByTestId, getByLabelText, "NI_File_unknown_error.sav");

        let panel = await getByTestId(/import=panel/i);
        expect(panel.textContent).toEqual("File Failed to Upload");
    });

    it("selects a file and an error is returned, display error with message", async () => {
        const {getByTestId, getByLabelText, getByText} = wrapper(render, PropsNI);

        await selectAndImportFile(getByTestId, getByLabelText, "NI_File_weird_error.sav");

        let panel = await getByTestId(/import=panel/i);
        expect(panel.textContent).toEqual("Error Occurred while Uploading File: Something strange has Occurred here");
    });

    test("that the websocket is called very 3 seconds", async () => {
        jest.useFakeTimers();

        server.on("connection", socket => {
            console.log("Server Connection");

            socket.on("message", function incoming(message) {
                console.log("Received from client:  %s", message);
                server.send({fileName: "file12", percent: 100, status: 2, errorMessage: ""});
            });
        });

        wrapper(render, Props);

        jest.advanceTimersByTime(1000);

        await server.connected;

        // Check the server has received the first request
        expect(server.messages.length).toEqual(1);

        jest.advanceTimersByTime(2500);

        // Check the server has received the second request from the setTimeout
        expect(server.messages.length).toEqual(2);

        jest.advanceTimersByTime(6000);

        expect(server.messages.length).toEqual(4);

        jest.useRealTimers();
    });
});