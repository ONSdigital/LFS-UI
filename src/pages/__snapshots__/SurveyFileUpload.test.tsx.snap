// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GB and NI survey file uploads renders the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <nav
          aria-label="Breadcrumb"
          class="breadcrumb"
          data-testid="breadcrumbs"
        >
          <ol
            class="breadcrumb__items"
          >
            <li
              class="breadcrumb__item u-fs-s"
              id="breadcrumb-1"
            >
              <a
                class="breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="breadcrumb__item u-fs-s"
              id="breadcrumb-2"
            >
              <a
                class="breadcrumb__link"
                href="/manage-batch/monthly/2019/1"
              >
                Manage Batch January 2019
              </a>
            </li>
            <li
              class="breadcrumb__item u-fs-s breadcrumb__item--current"
              data-testid="breadcrumb-current"
              id="breadcrumb-3"
            >
              Import Survey - Week 1
            </li>
          </ol>
        </nav>
        <h2>
          Import Survey
        </h2>
        <div>
          <dl
            aria-label="This is an example of the metadata component"
            class="metadata metadata__list grid grid--gutterless u-cf u-mb-l"
            title="This is an example of the metadata component"
          >
            <dt
              class="metadata__term grid__col col-2@m"
            >
              Survey
              :
            </dt>
            <dd
              class="metadata__value grid__col col-10@m"
              data-testid="metadata-test-0"
            >
              GB
            </dd>
            <dt
              class="metadata__term grid__col col-2@m"
            >
              Year
              :
            </dt>
            <dd
              class="metadata__value grid__col col-10@m"
              data-testid="metadata-test-1"
            >
              2019
            </dd>
            <dt
              class="metadata__term grid__col col-2@m"
            >
              Period
              :
            </dt>
            <dd
              class="metadata__value grid__col col-10@m"
              data-testid="metadata-test-2"
            >
              Week 1
            </dd>
          </dl>
        </div>
        <div
          style="width: 55%;"
        >
          <h4>
            Previous imports 
          </h4>
          <table
            class="table table--sortable"
            data-aria-asc="ascending"
            data-aria-desc="descending"
            data-aria-sort="Sort by"
            id="basic-table"
          >
            <thead
              class="table__head"
            >
              <tr
                class="table__row"
              >
                <th
                  class="table__header "
                  scope="col"
                >
                  Import Date
                </th>
                <th
                  class="table__header "
                  scope="col"
                >
                  Status
                </th>
              </tr>
            </thead>
            <tbody
              class="table__body "
            >
              <tr>
                <td
                  class="table__cell "
                  colspan="2"
                >
                  <div
                    class="panel panel--info panel--simple "
                  >
                    <div
                      class="panel__body"
                    >
                      <p>
                        Survey has not been previously imported
                      </p>
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="panel panel--info panel--simple "
          data-testid="import=panel"
          hidden=""
        >
          <div
            class="panel__body"
          >
            <p />
          </div>
        </div>
        <br />
        <form>
          <div
            class="field"
          >
            <p
              class="field"
            >
              <label
                class="label"
                for="U1"
              >
                Import GB File
                <br />
                <span
                  class="label__description"
                >
                  Only .sav accepted
                </span>
              </label>
              <input
                accept=".sav"
                class="input input--text input-type__input input--upload"
                id="U1"
                name="Upload 1"
                style="position: static;"
                type="file"
              />
            </p>
          </div>
          <br />
          <button
            class="btn field "
            data-testid="import-button"
            id="import-btn"
            type="button"
          >
            <span
              class="btn__inner"
            >
              Import
            </span>
          </button>
          <a
            class="field"
            href="/manage-batch/monthly/2019/1"
            style="margin-left: 15px;"
          >
            <button
              class="btn field btn--secondary "
              data-testid="button"
              type="button"
            >
              <span
                class="btn__inner"
              >
                Return to Manage Batch
              </span>
            </button>
          </a>
        </form>
        <br />
        <div
          hidden=""
        >
          <h4>
            Import Progress
          </h4>
          <div
            style="width: 60%;"
          >
            <table
              class="table table--sortable"
              data-aria-asc="ascending"
              data-aria-desc="descending"
              data-aria-sort="Sort by"
              id="basic-table"
            >
              <thead
                class="table__head"
              >
                <tr
                  class="table__row"
                >
                  <th
                    class="table__header "
                    scope="col"
                  >
                    File Name 
                  </th>
                  <th
                    class="table__header "
                    scope="col"
                  >
                    Step
                  </th>
                  <th
                    class="table__header "
                    scope="col"
                  >
                    Status
                  </th>
                </tr>
              </thead>
              <tbody
                class="table__body "
              >
                <tr
                  class="table__row nonSelectableTableRow"
                  data-testid="table-row"
                  tabindex="0"
                  title="Table Row"
                >
                  <td
                    class="table__cell "
                  />
                  <td
                    class="table__cell "
                  >
                    Import
                  </td>
                  <td
                    class="table__cell "
                  >
                    <span
                      class="status status--info"
                    />
                  </td>
                </tr>
              </tbody>
            </table>
            <div
              class="field"
            >
              <div
                class="progress-bar"
              >
                <div
                  class="progress-bar-filler"
                  style="width: 0%; background: rgb(136, 136, 136);"
                />
              </div>
            </div>
          </div>
        </div>
        <br />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <nav
        aria-label="Breadcrumb"
        class="breadcrumb"
        data-testid="breadcrumbs"
      >
        <ol
          class="breadcrumb__items"
        >
          <li
            class="breadcrumb__item u-fs-s"
            id="breadcrumb-1"
          >
            <a
              class="breadcrumb__link"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="breadcrumb__item u-fs-s"
            id="breadcrumb-2"
          >
            <a
              class="breadcrumb__link"
              href="/manage-batch/monthly/2019/1"
            >
              Manage Batch January 2019
            </a>
          </li>
          <li
            class="breadcrumb__item u-fs-s breadcrumb__item--current"
            data-testid="breadcrumb-current"
            id="breadcrumb-3"
          >
            Import Survey - Week 1
          </li>
        </ol>
      </nav>
      <h2>
        Import Survey
      </h2>
      <div>
        <dl
          aria-label="This is an example of the metadata component"
          class="metadata metadata__list grid grid--gutterless u-cf u-mb-l"
          title="This is an example of the metadata component"
        >
          <dt
            class="metadata__term grid__col col-2@m"
          >
            Survey
            :
          </dt>
          <dd
            class="metadata__value grid__col col-10@m"
            data-testid="metadata-test-0"
          >
            GB
          </dd>
          <dt
            class="metadata__term grid__col col-2@m"
          >
            Year
            :
          </dt>
          <dd
            class="metadata__value grid__col col-10@m"
            data-testid="metadata-test-1"
          >
            2019
          </dd>
          <dt
            class="metadata__term grid__col col-2@m"
          >
            Period
            :
          </dt>
          <dd
            class="metadata__value grid__col col-10@m"
            data-testid="metadata-test-2"
          >
            Week 1
          </dd>
        </dl>
      </div>
      <div
        style="width: 55%;"
      >
        <h4>
          Previous imports 
        </h4>
        <table
          class="table table--sortable"
          data-aria-asc="ascending"
          data-aria-desc="descending"
          data-aria-sort="Sort by"
          id="basic-table"
        >
          <thead
            class="table__head"
          >
            <tr
              class="table__row"
            >
              <th
                class="table__header "
                scope="col"
              >
                Import Date
              </th>
              <th
                class="table__header "
                scope="col"
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody
            class="table__body "
          >
            <tr>
              <td
                class="table__cell "
                colspan="2"
              >
                <div
                  class="panel panel--info panel--simple "
                >
                  <div
                    class="panel__body"
                  >
                    <p>
                      Survey has not been previously imported
                    </p>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="panel panel--info panel--simple "
        data-testid="import=panel"
        hidden=""
      >
        <div
          class="panel__body"
        >
          <p />
        </div>
      </div>
      <br />
      <form>
        <div
          class="field"
        >
          <p
            class="field"
          >
            <label
              class="label"
              for="U1"
            >
              Import GB File
              <br />
              <span
                class="label__description"
              >
                Only .sav accepted
              </span>
            </label>
            <input
              accept=".sav"
              class="input input--text input-type__input input--upload"
              id="U1"
              name="Upload 1"
              style="position: static;"
              type="file"
            />
          </p>
        </div>
        <br />
        <button
          class="btn field "
          data-testid="import-button"
          id="import-btn"
          type="button"
        >
          <span
            class="btn__inner"
          >
            Import
          </span>
        </button>
        <a
          class="field"
          href="/manage-batch/monthly/2019/1"
          style="margin-left: 15px;"
        >
          <button
            class="btn field btn--secondary "
            data-testid="button"
            type="button"
          >
            <span
              class="btn__inner"
            >
              Return to Manage Batch
            </span>
          </button>
        </a>
      </form>
      <br />
      <div
        hidden=""
      >
        <h4>
          Import Progress
        </h4>
        <div
          style="width: 60%;"
        >
          <table
            class="table table--sortable"
            data-aria-asc="ascending"
            data-aria-desc="descending"
            data-aria-sort="Sort by"
            id="basic-table"
          >
            <thead
              class="table__head"
            >
              <tr
                class="table__row"
              >
                <th
                  class="table__header "
                  scope="col"
                >
                  File Name 
                </th>
                <th
                  class="table__header "
                  scope="col"
                >
                  Step
                </th>
                <th
                  class="table__header "
                  scope="col"
                >
                  Status
                </th>
              </tr>
            </thead>
            <tbody
              class="table__body "
            >
              <tr
                class="table__row nonSelectableTableRow"
                data-testid="table-row"
                tabindex="0"
                title="Table Row"
              >
                <td
                  class="table__cell "
                />
                <td
                  class="table__cell "
                >
                  Import
                </td>
                <td
                  class="table__cell "
                >
                  <span
                    class="status status--info"
                  />
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="field"
          >
            <div
              class="progress-bar"
            >
              <div
                class="progress-bar-filler"
                style="width: 0%; background: rgb(136, 136, 136);"
              />
            </div>
          </div>
        </div>
      </div>
      <br />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
